hook.sh

```diff
@@ -279,18 +279,25 @@ while true; do
 
   display_warnings
 
-  # Ask the question (not using "read -p" as it uses stderr not stdout)
-  echo -en "${BLUE}Proceed with commit? [e/y/n/?] ${NC}"
+  if [ -t 1 ] || [ ! -z ${FAKE_TTY+x} ]; then
```

Comment:
`FAKE_TTY` is used by your tests to validate interactive behaviour when in fact there is no real tty. In order to be able to test the code we need a workaround for enabling the interactive. Look inside your test code for FAKE_TTY.  The ugly syntax is the **correct** way to check if a variable is defined in bash.